python/pyspark/sql/tests/test_udf.py

```diff
@@ -917,6 +923,7 @@ def test_complex_return_types(self):
         self.assertEqual(row[1], {"a": "b"})
         self.assertEqual(row[2], Row(col1=1, col2=2))
 
+    @unittest.skipIf(not have_pyarrow, pyarrow_requirement_message)
     def test_named_arguments(self):
```

Comment:
hm, I don't think this uses Arrow.

Reply:
Okay, let me check again.

Reply:
After testing, we found that the `assertDataFrameEqual ` method used in this UT requires it.

Reply:
@itholic can you make this test not requiring that method?

Reply:
Let me take a look. Do we want to fix it separately after this PR merging or before merging??

Reply:
FYI, in `assertDataFrameEqual` we just skip using pandas if it's not installed, but somehow it is not working properly for now. 

```python
    has_pandas = False
    try:
        # If pandas dependencies are available, allow pandas or pandas-on-Spark DataFrame
        import pyspark.pandas as ps
        import pandas as pd
        from pyspark.testing.pandasutils import PandasOnSparkTestUtils

        has_pandas = True
    except ImportError:
        # no pandas, so we won't call pandasutils functions
        pass
```

Anyway, I just create ticket SPARK-46821 to resolve this. I'm working on it.

Reply:
Hmm... it seems like actually `assertDataFrameEqual` is not depend on pandas??

```python
>>> import pandas
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'pandas'
>>> from pyspark.testing.utils import assertDataFrameEqual
>>> df1 = spark.range(10)
>>> df2 = spark.range(10)
>>> assertDataFrameEqual(df1, df2)
```

It works well without pandas. In the code, we skip using pandas if it's not installed:

```python
    has_pandas = False
    try:
        # If pandas dependencies are available, allow pandas or pandas-on-Spark DataFrame
        import pyspark.pandas as ps
        import pandas as pd
        from pyspark.testing.pandasutils import PandasOnSparkTestUtils

        has_pandas = True
    except ImportError:
        # no pandas, so we won't call pandasutils functions
        pass
```

> After testing, we found that the assertDataFrameEqual  method used in this UT requires it.

Did you test with uninstalling `pandas` clearly? we should uninstall `pandas-stubs` as well before uninstalling `pandas`. See https://github.com/apache/spark/pull/44745#issuecomment-1894850179 more detail.

Reply:
We just merged https://github.com/apache/spark/pull/44745 that bandaids for this issue. Could you try testing again after rebasing master?

Reply:
Sure, let me test it out now.

Reply:
@itholic From the test, it seems that there is still a problem:
https://github.com/panbingkun/spark/actions/runs/7634418170/job/20798310295#step:12:4443
<img width="998" alt="image" src="https://github.com/apache/spark/assets/15246973/bbbb52e0-2ed9-4694-8b35-0d4fd848b8e7">

Reply:
Oh.... seems like we got another problem that we try import `pyspark.pandas` in `assertDataFrameEqual`. Let me try to made a quick fix.

Reply:
Just realized that we're checking `pandas` and `pyarrow` when importing `pyspark.pandas` as below, and exit the program when it's not installed:

```python
try:
    require_minimum_pandas_version()
    require_minimum_pyarrow_version()
except ImportError as e:
    if os.environ.get("SPARK_TESTING"):
        warnings.warn(str(e))
        sys.exit()
    else:
        raise
```

@HyukjinKwon Maybe should we add some flag or something here to enable running test without `pandas` and `pyarrow`?

Reply:
yeah we can do

Reply:
Made a quick fix here: https://github.com/apache/spark/pull/44864

Reply:
After this is merged, I will rebase and test it again.